Packs/Zscaler/Integrations/Zscaler/Zscaler.py

```diff
@@ -1016,13 +1017,220 @@ def set_user_command(args):
+                ipv6_contents.append(content)
+            s_content = {
+                "IPv4": ipv4_contents,
+                "IPv6": ipv6_contents
+            }
+            hr = tableToMarkdown(f"IPv4 Destination groups ({len(ipv4_contents)})", ipv4_contents)
+            hr += tableToMarkdown(f"IPv6 Destination groups ({len(ipv6_contents)})", ipv6_contents)
+            entry = {
+                'Type': entryTypes['note'],
+                'Contents': s_content,
+                'ContentsFormat': formats['json'],
+                'ReadableContentsFormat': formats['markdown'],
+                'HumanReadable': hr,
+                'EntryContext': {'Zscaler.IPDestinationGroup': s_content},
+            }
+        else:
+            try:
+                cmd_url = "/ipDestinationGroups"
+                if exclude_type != '':
+                    cmd_url += f"?excludeType={exclude_type}"
+                responses = http_request('GET', cmd_url).json()
+            except json.decoder.JSONDecodeError as exp:
+                return_error(f'Failed to execute zscaler-list-ip-destination-group command. Error: {str(exp)}')
+            contents = list()
+            for response in responses:
+                content = {
+                    'ID': int(response.get('id', '')),
+                    'Name': response.get('name', ''),
+                    'Type': response.get('type', ''),
+                    'Description': response.get('description', ''),
+                    'Addresses': response.get('addresses', []),
+                    'IpCategories': response.get('ipCategories', []),
+                    'Countries': response.get('countries', [])
+                }
+                contents.append(content)
+            entry = {
+                'Type': entryTypes['note'],
+                'Contents': contents,
+                'ContentsFormat': formats['json'],
+                'ReadableContentsFormat': formats['markdown'],
+                'HumanReadable': tableToMarkdown(("IPv4 Destination groups"
+                                                  + f" ({len(contents)})"),
+                                                 contents),
+                'EntryContext': {'Zscaler.IPDestinationGroup': contents},
+            }
+    else:
+        try:
+            cmd_url = f"/ipDestinationGroups/{ip_group_id}"
+            response = http_request('GET', cmd_url).json()
+        except json.decoder.JSONDecodeError as exp:
+            return_error(f'Failed to execute zscaler-list-ip-destination-group command. Error: {str(exp)}')
+        content = {
+            'ID': int(response.get('id', '')),
+            'Name': response.get('name', ''),
+            'Type': response.get('type', ''),
+            'Description': response.get('description', ''),
+            'Addresses': response.get('addresses', []),
+            'IpCategories': response.get('ipCategories', []),
+            'Countries': response.get('countries', [])
+        }
```

Comment:
Similarly call the `get_contents` function to avoid repetition. 😸

Reply:
The final one returns a dictionary not a list, so can't be used again 👯‍♂️ I won this time.

Reply:
Technically you can still use the `get_contents` function and use list indexing to point to the `dict` object.

Reply:
Can we modify the `get_contents` function to handle both `list` and  `dict` responses?

Reply:
Keep in mind that you will also have to use type hints on the function definition.